src/video.js

```diff
@@ -64,3 +66,52 @@ export const checkVideoBidSetup = hook('sync', function(bid, adUnit, videoMediaT
 
   return true;
 }, 'checkVideoBidSetup');
+
+export function registerVASTTrackers(tracker) {
+  vastTrackers.push(tracker);
+}
+
+export function insertVastTrackers(trackers, vastXml, vastImpUrl) {
+  const doc = new DOMParser().parseFromString(vastXml, 'text/xml');
+  const wrappers = doc.querySelectorAll('VAST Ad Wrapper, VAST Ad InLine');
+  try {
+    if (wrappers.length) {
+      wrappers.forEach(wrapper => {
+        trackers['impressions'].forEach(trackingUrl => {
+          const impression = doc.createElement('Impression');
+          impression.appendChild(doc.createCDATASection(trackingUrl));
+          wrapper.appendChild(impression)
+        });
+      });
+      vastXml = new XMLSerializer().serializeToString(doc);
+    }
+  } catch (error) {
+    logError('an error happened trying to insert trackers in vastXml');
+  }
+  if (vastImpUrl && vastImpUrl.length > 0) {
+    trackers['impressions'].forEach(trackingUrl => {
+      if ((trackingUrl + encodeURI(vastImpUrl)).length < 2048) {
```

Comment:
I am not sure how this works in general. from the unit tests I see the idea - the tracker URL ends in `&redir=` and presumably that means the server will reply with a redirect. Is that standardized - what's stopping me from registering a tracker that doesn't act as an open redirect? (isn't it also a security risk for you to run an open redirect service?)

Reply:
That's the idea. It's not standardize but tracking video impression in prebid is new so it could be. This is something which was done before this advertising click command in order to track click count and it worked with an open redirect service. It's just an URL doing some http 302 redirect. 

It would be a security risk if the domain use for this redirect was from some trusted source  such as a well known website users are used to browse to. This should not be the case with analytics modules which would use this feature. If you have any other idea about how this could be improved I am open to make it another way. The best of could would be for vastImpUrl to be an array of strings instead of a string, but it would break existing modules and videoCache too I guess.

Reply:
I don't think we want to encourage Prebid modules to generate redirects to redirects as standard practice; besides the URL length limitation, each redirect also adds a network roundtrip that the end user has to wait for before being shown their ad. 

It might be OK for your particular module to do this (without support from core, as you should be able to do now - since this would only matter when there's no cache). 

As for how to do this in general - I don't know if it works, but an idea could be to generate a VAST wrapper to `vastURL` containing the additional trackers, and package it in a [data URL](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs) that replaces the original `vastURL`.

Reply:
It was brought to my attention that a video player loading the VAST does not mean that you got the impression - playback could stop before it's time to show the ad. This is another reason to not use this technique from core, even if for your particular module you may be OK with the discrepancy.

For this PR I think it'd also be fine to only inject trackers when there is a cache.

Reply:
My problem here is that videoCache even caches vastUrl and vastImpUrl if I didn't misinterpreted these lines :

https://github.com/prebid/Prebid.js/blob/822c09e5e10077528431054e90073bc4d58099f7/src/videoCache.js#L45-L70

If the bid contains both vastXml and vastUrl, vastXml will be cached and this will take precedence over vastUrl in `src/auctions.js`
But if only vastUrl is available, this vastUrl is translated to a vast through `wrapURI` func and then cached. So vastUrl impressions can't be tracked. 
Maybe then I would need to add modifications to `src/videoCache.js` in order to inject additional trackers by `wrapURI` then if that's okay for you ?

Reply:
You can inject your trackers just before the VAST is to be cached, so that it doesn't matter if it was the "real" one or the wrapper generated by the caching logic; and ignore the no-cache case (bids can have a `vastXml` only when cache is enabled). I think that's an OK strategy.

Reply:
I've made changes not to update vastImpUrl anymore and add trackers in `wastURI` function.